addons/web/static/src/model/relational_model/record.js

```diff
@@ -384,13 +390,22 @@ export class Record extends DataPoint {
 
     _computeDataContext() {
         const dataContext = {};
+        const one2manyContext = {
+            withVirtualIds: {},
+            withoutVirtualIds: {},
+        };
         const data = toRaw(this.data);
         for (const fieldName in data) {
             const value = data[fieldName];
             const field = this.fields[fieldName];
             if (["char", "text", "html"].includes(field.type)) {
                 dataContext[fieldName] = this._textValues[fieldName];
-            } else if (["one2many", "many2many"].includes(field.type)) {
+            } else if (field.type === "one2many") {
+                one2manyContext.withVirtualIds[fieldName] = value.currentIds;
+                one2manyContext.withoutVirtualIds[fieldName] = value.currentIds.filter(
+                    (id) => typeof id === "number"
+                );
+            } else if (field.type === "many2many") {
```

Comment:
I think I would handle both m2m and o2m  the same way, because you can have a widget one2many on a many2many field, which I think can then have virtual records